build.cmd

```
@@ -382,18 +382,41 @@ echo MYGET_APIKEY=%MYGET_APIKEY%
-)
-if "%VS150COMNTOOLS%" EQU "" if exist "%ProgramFiles(x86)%\Microsoft Visual Studio\2017\Professional\Common7\Tools\VsDevCmd.bat" (
+))
+if [""] == ["%VS150COMNTOOLS%"] (if exist "%ProgramFiles(x86)%\Microsoft Visual Studio\2017\Professional\Common7\Tools\VsDevCmd.bat" (
     call "%ProgramFiles(x86)%\Microsoft Visual Studio\2017\Professional\Common7\Tools\VsDevCmd.bat"
-)
-if "%VS150COMNTOOLS%" EQU "" if exist "%ProgramFiles(x86)%\Microsoft Visual Studio\2017\Community\Common7\Tools\VsDevCmd.bat" (
+))
+if [""] == ["%VS150COMNTOOLS%"] (if exist "%ProgramFiles(x86)%\Microsoft Visual Studio\2017\Community\Common7\Tools\VsDevCmd.bat" (
     call "%ProgramFiles(x86)%\Microsoft Visual Studio\2017\Community\Common7\Tools\VsDevCmd.bat"
-)
-if "%VS150COMNTOOLS%" EQU "" if exist "%ProgramFiles(x86)%\Microsoft Visual Studio\2017\BuildTools\Common7\Tools\VsDevCmd.bat" (
+))
+if [""] == ["%VS150COMNTOOLS%"] (if exist "%ProgramFiles(x86)%\Microsoft Visual Studio\2017\BuildTools\Common7\Tools\VsDevCmd.bat" (
     call "%ProgramFiles(x86)%\Microsoft Visual Studio\2017\BuildTools\Common7\Tools\VsDevCmd.bat"
+))
+
+
+echo aaa !link_exe!
+if [""] == ["!link_exe!"] (if exist "%VCToolsInstallDir%bin\HostX64\x86\link.exe" (
+    set link_exe="%VCToolsInstallDir%bin\HostX64\x86\link.exe"
+))
+echo bbb !link_exe!
+if [""] == ["!link_exe!"] (
+    echo did not find %VCToolsInstallDir%bin\HostX64\x86\link.exe
+    dir "%VCToolsInstallDir%bin\HostX64\x86"
+    set link_exe="%~dp0packages\VisualCppTools.14.0.24519-Pre\lib\native\bin\link.exe"
+    if not exist "%~dp0packages\VisualCppTools.14.0.24519-Pre\lib\native\bin\link.exe" (
+        %_nugetexe% install -source https://www.myget.org/F/fsharp-daily/api/v3/index.json VisualCppTools -version 14.0.24519-Pre -out packages
+    )
+)
+echo ccc !link_exe!
+if not exist "!link_exe!" (
+    echo Error: failed to find link.exe
+    goto :failure
 )
+echo ddd !link_exe!
+goto :failure
```

The section above is just a copy to fail faster.